File: frontend/src/features/professional/hooks/GetCurrentProfessional/GetCurrentProfessionalDto.ts
import type { Address } from "@/features/auth";
import type { VerificationStatus } from "@/features/admin/types/adminTypes";
import type { ServiceDto } from "@/features/professional/hooks/GetServicesBySpecialization";
import type { SpecializationDto } from "@/features/professional/hooks/GetSpecializations";

export interface GetCurrentProfessionalDto {
  id: string;
  userId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;
  gender: string;
  address: Address;
  specialization: SpecializationDto;
  services: ServiceDto[];
  experience: number;
  visitPrice: number;
  bio: string;
  profilePictureUrl: string;
  verificationStatus: VerificationStatus;
}

File: backend/Modules/Professionals/Modules.Professionals.Features/Auth/GetCurrentProfessional/GetCurrentProfessionalDto.cs
using Modules.Common.Features.ValueObjects;
using Modules.Professionals.Domain.Enums;
using Modules.Professionals.Features.GetServices;
using Modules.Professionals.Features.GetSpecializations;

namespace Modules.Professionals.Features.Auth.GetCurrentProfessional;

public sealed record GetCurrentProfessionalDto(
    Guid Id,
    Guid UserId,
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    string DateOfBirth,
    string Gender,
    Address Address,
    SpecializationDto Specialization,
    List<ServiceDto> Services,
    int Experience,
    int VisitPrice,
    string? Bio,
    string? ProfilePictureUrl,
    VerificationStatus VerificationStatus);
